ci/cron/src/Main.hs

```diff
@@ -220,11 +230,12 @@ build_docs_folder path versions latest = do
             -- Not going through Aeson because it represents JSON objects as
             -- unordered maps, and here order matters.
             let versions_json = versions
+                                & map name
```

Comment:
I’m struggling to understand what changed to require this now. Can you help me out?

Reply:
The script used to decide whether to run or not based on whether the most recent published release had changed. If we want to publish documentation for snapshot releases that won't work anymore.

Because we do not want prereleases in the drop-down, they cannot be in versions.json, but we still need to know which ones we have published already, so we need to keep track of them _somewhere_.

We also need to know, for each docs version we have built, whether we built it as prerelease or not, because if a version changes (we publish a stable release after this has run), we still need to rerun.

This is why we need this snapshots.json, and why it is fetched by `fetch_s3_versions`.

Reply:
:man_facepalming: Thanks for the explanation!

Reply:
I should probably type my subfunctions. It would have made the change much clearer, and @nickchapman-da would stop telling me to do it. I'll add type annotations next time I need to touch this script.

Reply:
Yeah. More types!